x/twap/store.go

```diff
@@ -93,6 +95,8 @@ func (k Keeper) pruneRecordsBeforeTimeButNewest(ctx sdk.Context, lastKeptTime ti
 	}
 	seenPoolAssetTriplets := map[uniqueTriplet]struct{}{}
```

Comment:
Aw yikes, were going to be repeating a lot of reads every block here :thinking: . 1 read per pool, so potentially 1400 extra iteration reads.

Reply:
Trying to think of any clever thing here, aside from:
- maintaining "start key" + seen triplets (which would work, with one big read rather than 1400 small reads which is worse)

Reply:
We can restructure the iterations / keys which would help in the future!

Just realized there is a more "dumb" short term fix for this. We can iterate

Reply:
Added issue to restructure keys for later https://github.com/osmosis-labs/osmosis/issues/7435